app/scripts/controller-init/confirmations/ppom-controller-init.test.ts

```diff
@@ -0,0 +1,66 @@
+import { PPOMController } from '@metamask/ppom-validator';
+import { buildControllerInitRequestMock, CHAIN_ID_MOCK } from '../test/utils';
+import { PPOMControllerInit } from './ppom-controller-init';
+
+type PPOMControllerOptions = ConstructorParameters<typeof PPOMController>[0];
+
+jest.mock('@metamask/ppom-validator');
+
+function buildInitRequestMock() {
+  const requestMock = buildControllerInitRequestMock();
+
+  // @ts-expect-error Mocked subset of full state object
+  requestMock.getController.mockReturnValue({
+    state: { securityAlertsEnabled: true },
+  });
+
+  return requestMock;
+}
+
+describe('PPOM Controller Init', () => {
+  const ppomControllerClassMock = jest.mocked(PPOMController);
+
+  function testConstructorProperty<T extends keyof PPOMControllerOptions>(
```

Comment:
Nit: I struggled a bit in understanding what a "constructor property" was. I see that it's referring to a parameter of the "options" object passed to the constructor. Maybe we could make this more clear by calling this `testConstructorOption` or `testConstructorParameter`? This would also help differentiate the thing we're checking here from the `controllerProperties` passed in (at first I thought these were for the same controller).

A JSDoc comment would also be very helpful here in understanding what the tests are asserting

Reply:
Renamed and comment added.